src/octoprint/util/json/serializing.py

```diff
@@ -0,0 +1,156 @@
+__license__ = "GNU Affero General Public License http://www.gnu.org/licenses/agpl.html"
+__copyright__ = "Copyright (C) 2022 The OctoPrint Project - Released under terms of the AGPLv3 License"
+
+import datetime
+import json
+import time
+from collections import OrderedDict
+from typing import Any, Callable, Dict, List
+
+from frozendict import frozendict
+
+from octoprint.util import to_bytes, to_unicode
+
+from .encoding import JsonEncoding
+
+
+class SerializableJsonEncoding(JsonEncoding):
```

Comment:
I like this separation. I was filling iffy before about how the previous interface filled too many goals in one place, but this is a lot better.